File: apps/frontend/src/types/audit-log.ts
// src/types/audit-log.ts
import type { Pagination } from "./api";

export interface AuditLogEntry {
  id: number;
  created_at: string;
  actor_id: string | null;
  actor_email: string | null;
  table_name: string;
  record_id: string;
  action: string;
  changes: Record<string, unknown>;
  ip_address: string | null;
  user_agent: string | null;
  notes: string | null;
}

export interface PaginatedAuditLog {
  items: AuditLogEntry[];
  pagination: Pagination;
}

File: apps/backend/app/Schemas/admin/audit_log.py
# apps/backend/app/Schemas/admin/audit_log.py
from datetime import datetime
from typing import Any
from uuid import UUID
from pydantic import BaseModel, ConfigDict


class AuditLogEntry(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    created_at: datetime
    actor_id: UUID | None
    actor_email: str | None
    table_name: str
    record_id: str
    action: str
    changes: dict[str, Any]
    ip_address: str | None
    user_agent: str | None
    notes: str | None
